compiler/fuzzer/src/fuzzer.rs

```diff
@@ -113,23 +124,25 @@ impl Fuzzer {
         );
         trace!("{}", result.to_string(&call_string));
         match result {
-            RunResult::Timeout => self.create_new_fuzzing_case(pool),
+            RunResult::Timeout => self.create_new_fuzzing_case(pool, total_coverage),
             RunResult::Done { .. } | RunResult::NeedsUnfulfilled { .. } => {
-                // TODO: For now, our "coverage" is just the number of executed
-                // instructions. In the future, we should instead actually look
-                // at what parts of the code ran. This way, we can boost inputs
-                // that achieve big coverage with few instructions.
-                let coverage = runner.num_instructions;
-
-                // We favor small inputs with good code coverage.
-                let score = {
-                    let coverage = coverage as Score;
-                    let complexity = complexity_of_input(&runner.input) as Score;
-                    let score: Score = 0.1 * coverage - 0.4 * complexity;
-                    score.clamp(0.1, Score::MAX)
-                };
-                pool.add(runner.input, score);
-                self.create_new_fuzzing_case(pool)
+                if total_coverage
+                    .relative_coverage_of_range(self.lir.range_of_function(&self.function_id))
+                    == 1.0
+                {
+                    Status::TotalCoverageButNoPanic
+                } else {
+                    // We favor small inputs with good code coverage.
+                    let score = {
+                        let complexity = complexity_of_input(&runner.input) as Score;
+                        let score: Score = 0.1
+                            * runner.coverage.improvement_on(&total_coverage) as f64
```

Comment:
If we only want to achieve full coverage of the function itself and don't care about the callees (the currently implemented goal), then this should also use just the coverage improvement of the current function and not the global coverage improvement